panther_utils/include/panther_utils/common_utilities.hpp

```diff
@@ -40,6 +41,28 @@ std::map<std::string, T> PrefixMapKeys(
   return prefixed_map;
 }
 
+/**
+ * @brief Opens a file with the specified mode.
+ *
+ * This function opens a file with the specified mode and returns a std::fstream object.
+ * If the file fails to open, it throws a std::runtime_error.
+ *
+ * @param file_path The path to the file to be opened.
+ * @param mode The mode in which the file should be opened (e.g., std::ios_base::in |
+ * std::ios_base::out).
+ * @return std::fstream The opened file stream.
+ *
+ * @throws std::runtime_error if the file fails to open.
+ */
+std::fstream OpenFile(const std::string & file_path, const std::ios_base::openmode & mode)
+{
+  std::fstream file(file_path, mode);
+  if (!file.is_open()) {
+    throw std::runtime_error("Failed to open file: " + file_path);
+  }
+  return file;
+}
+
```

Comment:
Please add the corresponding unit test to common_utilities.hpp

Reply:
added